front/app/containers/Admin/projects/project/nativeSurvey/utils.tsx

```diff
@@ -1,13 +1,61 @@
-// Services
+import React from 'react';
+
+// services
+import { IFlatCustomField } from 'services/formCustomFields';
 import { IPhaseData, updatePhase } from 'services/phases';
 import { IProjectData, updateProject } from 'services/projects';
 
-// Typings
+// typing
 import { Multiloc } from 'typings';
 
-// Utils
+// utils
 import { isNilOrError } from 'utils/helperUtils';
 
+// components
+import DeleteFormResultsNotice from './DeleteFormResultsNotice';
+import { FormBuilderConfig } from 'components/FormBuilder/utils';
+
+// intl
+import messages from './messages';
+
+export const getUpdatedConfiguration = (
+  config: FormBuilderConfig,
+  formCustomFields: IFlatCustomField[] | undefined | Error
+) => {
+  config.formCustomFields = formCustomFields;
+  return config;
+};
+
+export const nativeSurveyConfig: FormBuilderConfig = {
+  formBuilderTitle: messages.survey,
+  viewFormLinkCopy: messages.viewSurvey,
+  toolboxTitle: messages.addSurveyContent,
+  formSavedSuccessMessage: messages.successMessage,
+  supportArticleLink: messages.supportArticleLink,
+  formEndPageLogicOption: messages.surveyEnd,
+  questionLogicHelperText: messages.questionLogicHelperText,
+  pagesLogicHelperText: messages.pagesLogicHelperText,
+
+  toolboxFieldsToExclude: [],
+  formCustomFields: undefined,
+
+  showStatusBadge: true,
+  isLogicEnabled: true,
+  isEditPermittedAfterSubmissions: false,
+
+  getGoBackUrl: (projectId: string) => {
```

Comment:
Minor: We could make this a URL string and not a function. In this case, we know that we need the `projectId` but we can't be sure that that will always be the case or that we won't need more parameters. The component passing in the config can handle that logic and merely pass in the string

Reply:
What would you suggest the config itself provides then? Just the last part of the URL? Or should I not pass in anything in the config itself, but declare that goBackURL exists in the config, and set it's value for the first time in the component passing in the config?

Currently I have:
`  getGoBackUrl: (projectId: string) => {
    return '/admin/projects/${projectId}/native-survey';
  },`
But I could have:
  `goBackUrl: 'native-survey',` (And then handle the actual URL creation in the component passing in the config)
Or I could not set anything here and just assign the variable in the component passing in the config.

Reply:
We could pass the entire url. The component passing the config will know the project and any other information and as such, it can create the URL

Reply:
If I have `goBackUrl: '/admin/projects/${projectId}/native-survey'` in the config itself though, it will complain since projectId isn't known yet :thinking: While the component passing the config will have projectId, I don't think I can pass that whole URL  in via the config if there isn't a projectId variable? Or am I missing something?

Reply:
The projectId is in the `SurveyFormBuilder` component. We currently have a `getUpdatedConfiguration`. This method can pass in the `projectId` that is used in constructing the url. The`nativeSurveyConfig` could be a returned object from a function that takes in any parameters for attributes that we want to be dynamic. This way each usage handles it's own logic. It is minor though so we could skip and change later if we need to

Reply:
Ahh ok, yep agreed! :) I misunderstood what you were saying about passing in the entire URL. Handling it in the getUpdatedConfiguration was what I was thinking as well.

Reply:
Something like

```
export const getSurveyConfig = (
  projectId: string,
  formCustomFields: IFlatCustomField[] | undefined | Error
): FormBuilderConfig => {

  return {
    formBuilderTitle: messages.survey,
    viewFormLinkCopy: messages.viewSurvey,
    toolboxTitle: messages.addSurveyContent,
    formSavedSuccessMessage: messages.successMessage,
    supportArticleLink: messages.supportArticleLink,
    formEndPageLogicOption: messages.surveyEnd,
    questionLogicHelperText: messages.questionLogicHelperText,
    pagesLogicHelperText: messages.pagesLogicHelperText,
  
    toolboxFieldsToExclude: [],

    formCustomFields,
  
    showStatusBadge: true,
    isLogicEnabled: true,
    isEditPermittedAfterSubmissions: false,
  
    goBackUrl: `/admin/projects/${projectId}/native-survey`,
    getDeleteFormResultsNotice: (projectId: string) => {
      return (
        <DeleteFormResultsNotice
          projectId={projectId}
          redirectToSurveyPage={true}
        />
      );
    },
  }
};
```